EventsExpress/ClientApp/src/services/EventsExpressService.js

```diff
@@ -371,26 +408,91 @@ export default class EventsExpressService {
-                'Content-Type': 'application/json',
-                'Authorization': `Bearer ${localStorage.getItem('token')}`
-            }),
-            body: JSON.stringify(data)
+    setResource = async (url, data) => {
+        const res = await fetch(
+            this._baseUrl + url,
+            {
+                method: "post",
+                headers: new Headers({
+                    'Content-Type': 'application/json',
+                    'Authorization': `Bearer ${localStorage.getItem('token')}`
+                }),
+                body: JSON.stringify(data)
+            }
+        );
+        if (res.ok) {
+            return res;
         }
-    );
+        else {
+            if (res.status === 401) {
+                let setResUpdateToken = null;
+                localStorage.removeItem("token");
 
-    setResourceWithData = (url, data) => fetch(
-        this._baseUrl + url,
-        {
-            method: "post",
-            headers: new Headers({
-                'Authorization': `Bearer ${localStorage.getItem('token')}`
-            }),
-            body: data
+                setResUpdateToken = await fetch('api/token/refresh-token', {
+                    method: "POST"
+                });
+
+                if (!setResUpdateToken.ok) {
+                    return this.getResourceResponseHandler(setResUpdateToken);
+                }
+                let result = await this.getResourceResponseHandler(setResUpdateToken);
+                localStorage.setItem('token', result.jwtToken);
+
+                return this.setResource(url, data);
+
+            }
+            else {
+                return {
+                    error: {
+                        ErrorCode: res.status,
+                        massage: await res.statusText
+                    }
+                }
+            }
+        }
+
+    }
+
+    setResourceWithData = async (url, data) => {
+        const res = await fetch(
+            this._baseUrl + url,
+            {
+                method: "post",
+                headers: new Headers({
+                    'Authorization': `Bearer ${localStorage.getItem('token')}`
+                }),
+                body: data
+            }
+        );
+        if (res.ok) {
+            return res;
         }
-    );
+        else {
+            if (res.status === 401) {
+                let setResWithDataUpdateToken = null;
+                localStorage.removeItem("token");
+
+                setResWithDataUpdateToken = await fetch('api/token/refresh-token', {
+                    method: "POST"
+                });
+
+                if (!setResWithDataUpdateToken.ok) {
+                    return this.getResourceResponseHandler(setResWithDataUpdateToken);
+                }
+                let result = await this.getResourceResponseHandler(setResWithDataUpdateToken);
+                localStorage.setItem('token', result.jwtToken);
+
+                return this.setResourceWithData(url, data);
```

Comment:
DRY !!!

This should be moved into separate method.
You can use callback parameter for last return call

Reply:
created functions RefreshHandler and ErrorHandler and used in this case